### DISCUSSION TITLE
The size of files that are too big or binary isn't displayed.

### DISCUSSION THREAD

**Discussion by AYMENJD** [community/community#46179] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

This issue only exists on beta version of `Code Search and navigation`:
![MessagesManager](https://user-images.githubusercontent.com/53928879/216415744-c17dbe97-729f-462e-8a2f-61672f691172.png)

**Comment by MostafaWaleed0** [comment]
@AYMENJD 

GitHub has a limit on the size of files that can be displayed in the web interface. This limit is set to 100 MB for individual files, and repository size is limited to 2 GB. If a file is larger than this limit, it will not be displayed in the web interface and cannot be viewed or downloaded directly from GitHub.

For large binary files such as images, audio, and video files, GitHub recommends using a Git Large File Storage (LFS) extension. Git LFS allows you to store and manage large files directly in your Git repository, rather than in the Git history. This helps reduce the repository size and improve performance when cloning and checking out repositories.

If you need to share large files that are too big for GitHub, you can consider using other file-sharing services, such as Google Drive, Dropbox, or WeTransfer. You can also use GitHub Releases to publish and distribute large files associated with your project, such as binary builds or installers.

In summary, if you need to store and share large binary files, you can use Git LFS or other file-sharing services, but large files will not be displayed in the web interface of GitHub.

**Reply by AYMENJD** [reply]
I don't mean that. As i wrote in the title, in the beta version of github.com you can't see the file size of the big file or binary. How you want the user to view it as raw without informing the user about the actual file size?

**Reply by MostafaWaleed0** [reply]
In GitHub, the size of large files or binary files may not be displayed for performance reasons. This is because large files can slow down the loading speed of the page and affect the overall user experience.

To work around this limitation, you can try the following options:

1. Use the Git command line: You can use the `git ls-remote --size <repository>` command to see the size of the files in a repository. The command will return the sizes of all the objects in the repository, including large files and binaries.

2. Use a third-party tool: There are several third-party tools available that can help you view the sizes of files in a GitHub repository, such as Git-LFS (Large File Storage). These tools can help you manage large files and binaries in your repository while still keeping track of their sizes.

Keep in mind that the actual size of a file may be different from what is displayed in GitHub, due to the way binary files are stored in Git and other factors.

**Reply by AYMENJD** [reply]
> In GitHub, the size of large files or binary files may not be displayed for performance reasons. This is because large files can slow down the loading speed of the page and affect the overall user experience.

The non beta version of github.com displays it:

![MessagesManager](https://user-images.githubusercontent.com/53928879/216548842-a4e3edcd-1dca-4e5d-a030-11bba01b9eb7.png)

**Reply by MostafaWaleed0** [reply]
If the solutions don't work, This is a known issue that has been reported and is being addressed by GitHub's development team. I suggest checking the [GitHub blog](https://github.blog/) or forums for updates on when this issue will be resolved.

**Comment by jsoref** [comment]
Fwiw, this _appears_ to be fixed.

https://github.com/check-spelling-sandbox/large-files-0/blob/main/50m.txt
<img width="788" alt="image" src="https://github.com/community/community/assets/2119212/f2e95fa9-24e2-4db6-8591-3404916be2a0">

https://github.com/check-spelling-sandbox/large-files-0/blob/main/98m.txt
<img width="791" alt="image" src="https://github.com/community/community/assets/2119212/2da23ae1-eb2a-4e08-9e83-e8d372a86021">

```sh
for u in {b,k,m}; do for s in $(for a in $(seq 10); do for b in $(seq 10); do for c in $(seq 10) ; do echo $(( $a * $b * $c )); done; done; done|sort -uh); do truncate -s $s$u $s$u.txt; done; done
rm ???m.txt 1000m.txt
git add *
git commit -m 'test'
```
